datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -276,6 +377,7 @@ macro_rules! binary_array_op_scalar {
             DataType::Date64 => {
                 compute_op_scalar!($LEFT, $RIGHT, $OP, Date64Array)
             }
+            DataType::Boolean => compute_bool_op_scalar!($LEFT, $RIGHT, $OP, BooleanArray),
```

Comment:
adding this line and the one below it adds all the new support, which is kind of cool! It is terrifying how many functions end up being called :)